### PR TITLE
[MINT-3106] Pre-launch Cleanup

### PR DESCRIPTION
**Pull request by garyjzhao**
# MINT-3106

### Acceptance Criteria

- Models you're following table will need to update FFS CRs and TDLs
- Models with FFS CRs and TDLs will need to be updated
- Remove the ability to add a CR/TDL in MINT
- Remove FFS CR/TDL section from the Model Plan task list
- Remove FFS CR/TDLs from the CSV file

### Additional Details

- [Figma](https://www.figma.com/design/qXhO12xSSkDF4LaToGKJoV/MINT-Blue-Sky-Ideas?node-id=2852-95648&t=NOqkTpmuDmBfY6dk-4): Adjustments to existing task list
- [Figma](https://www.figma.com/design/qXhO12xSSkDF4LaToGKJoV/MINT-Blue-Sky-Ideas?node-id=2852-97442&t=NOqkTpmuDmBfY6dk-4): Clean up to existing tables

## How to test this change

1.  Verify that you cannot add a new CR and/or TDL
2. Task list to match figma
3. Favorite Model Plans have the correct CR/TDL shown
4. Models with FFS CRs and TDLs table have correct CR/TDL shown
5. correct CR/TDL shown in read only view

## PR Author Checklist

- [x] I have provided a detailed description of the changes in this PR.
- [x] I have provided clear instructions on how to test the changes in this PR.
- [ ] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: src/gql/operations/ModelPlan/GetFavorites.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ export default gql(/* GraphQL */ `
...
       }
+      echimpCRsAndTDLs {
+        ... on EChimpCR {
+          id
+        }
+        ... on EChimpTDL {
+          id
+        }
+      }
       crs @include(if: $isMAC) {
```
Comment (ClayBenson94) [inline src/gql/operations/ModelPlan/GetFavorites.ts:33]: Like here!

File: src/gql/operations/ModelPlan/GetModelPlans.ts [hunk 2]
```diff
@@ -48,6 +48,14 @@ export default gql(/* GraphQL */ `
...
       }
+      echimpCRsAndTDLs {
+        ... on EChimpCR {
+          id
+        }
+        ... on EChimpTDL {
+          id
+        }
+      }
       crs @include(if: $isMAC) {
```
Comment (ClayBenson94) [inline src/gql/operations/ModelPlan/GetModelPlans.ts:59]: or here 👀

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]
These changes look good!

One thought I had: It might be a good idea to fully remove all instances where we query `crs` or `tdls` (the old types) in our GQL queries so we can 100% assert that there's no lingering references (in mock types _and_ in generated types / app code) 

Would you mind taking a crack at that in this PR? If we've got all the right queries updated already, it should basically be a no-op (minus maybe some generated types) and will make removing these on the backend all the easier when we go to do that work post-deploy!

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
Getting some testing warning identifying some incorrect mocks that have to do with the isMAc filter changes

```
More info: https://sass-lang.com/d/legacy-js-api
stderr | src/features/Home/index.test.tsx > The home page > renders empty message for no settings selected
No more mocked responses for the query: query GetFavorites($filter: ModelPlanFilter!) {
  modelPlanCollection(filter: $filter) {
    id
    modelName
    isFavorite
    nameHistory(sort: DESC)
    isCollaborator
    status
    basics {
      id
      goal
      performancePeriodStarts
    }
    collaborators {
      id
      userAccount {
        id
        commonName
      }
      teamRoles
    }
    echimpCRsAndTDLs {
      ... on EChimpCR {
        id
      }
      ... on EChimpTDL {
        id
      }
    }
  }
}

Expected variables: {"filter":"INCLUDE_ALL"}

Failed to match 1 mock for this query. The mocked response had the following variables:
  {"filter":"INCLUDE_ALL","isMAC":true}
```

There's also some query errors on the echimp card we want to address
https://www.apollographql.com/docs/react/errors#%7B%22version%22%3A%223.11.8%22%2C%22message%22%3A12%2C%22args%22%3A%5B%22initiator%22%2C%22%7B%5Cn%20%20%5C%22__typename%5C%22%3A%20%5C%22EChimpCR%5C%22%2C%5Cn%20%20%5C%22id%5C%22%3A%20%5C%22123%5C%22%2C%5Cn%20%20%5C%22title%5C%22%3A%20%5C%22Echimp%20CR%5C%22%2C%5Cn%20%20%5C%22emergencyCrFlag%5C%22%3A%20true%2C%5Cn%20%20%5C%22sensitiveFlag%5C%22%3A%20false%2C%5Cn%20%20%5C%22crStatus%5C%22%3A%20%5C%22Open%5C%22%2C%5Cn%20%20%5C%22implementationDate%5C%22%3A%20%5C%222022-07-30T05%3A00%3A00Z%5C%22%5Cn%7D%22%5D%7D

Also getting mock warning on ShareExport
```
[Network error]: Error: No more mocked responses for the query: query GetEchimpCrandTDL($id: UUID!) {
  modelPlan(id: $id) {
    echimpCRsAndTDLs {
      ... on EChimpCR {
        id
        title
        emergencyCrFlag
        sensitiveFlag
        crStatus
        implementationDate
        initiator
        crSummary {
          rawContent
        }
        relatedCrTdlNumbers
      }
      ... on EChimpTDL {
        id
        title
        issuedDate
      }
    }
  }
}

Expected variables: {"id":"f11eb129-2c80-4080-9440-439cbe1a286f"}
```

Mock errors on more echimp stuff
https://go.apollo.dev/c/err#%7B%22version%22%3A%223.11.8%22%2C%22message%22%3A12%2C%22args%22%3A%5B%22initiator%22%2C%22%7B%5Cn%20%20%5C%22__typename%5C%22%3A%20%5C%22EChimpCR%5C%22%2C%5Cn%20%20%5C%22id%5C%22%3A%20%5C%22123%5C%22%2C%5Cn%20%20%5C%22title%5C%22%3A%20%5C%22Echimp%20CR%5C%22%2C%5Cn%20%20%5C%22emergencyCrFlag%5C%22%3A%20true%2C%5Cn%20%20%5C%22sensitiveFlag%5C%22%3A%20false%2C%5Cn%20%20%5C%22crStatus%5C%22%3A%20%5C%22Open%5C%22%2C%5Cn%20%20%5C%22implementationDate%5C%22%3A%20%5C%222022-07-30T05%3A00%3A00Z%5C%22%5Cn%7D%22%5D%7D


All of these errors should be caught in the editor.  You should type your mocks.  They aren't typed here:

https://github.com/CMS-Enterprise/mint-app/blob/552f3274c808be6108609616cb6cb1b6931b0c57/src/features/ModelPlan/CRTDL/CRTDLs/index.test.tsx#L16-L30

something like
`type GetEchimpCrandTdlQueryType = GetEchimpCrandTdlQuery['modelPlan'];`

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
There's still some mock errors with echimp / shareExport.  There's a few other unit tests that also have some mock errors.  One (prepareForClearance) is known, and is fine with the error.  The other one for IT solutions can be addressed, but it's tech debt and we will soon fade out that feature

Screenshot 2024-10-21 at 3 48 12 PM

The reason why we want to generally address these, is that they can affect getting an accurate representation of the async component, and snapshots

**Review by patrickseguraoddball — APPROVED** [review]
LG , thanks!

**Pull request merged**